src/vaults/Vault.ts

```diff
@@ -235,19 +279,25 @@ export class Vault {
     }
 
     async loadMetadata() {
+        let contentJson;
+        let metaContainer: object = {};
+        let compressedContainerSize: number = 0;
+        let deCompressedContainerSize: number = 0;
         try {
             const data = await this.getFile(Vault.METADATA_FILE_NAME);
             this.meta = await JSON.parse(data);
             this.containers = {};
             for (const name in this.meta.containers) {
-                this.containers[name] = Container.typeFactory(
-                    this.meta.containers[name].container_type,
-                    this,
-                    name,
-                    this.meta.containers[name],
-                );
+                compressedContainerSize = compressedContainerSize + this.meta.containers[name].length;
+                contentJson = await this.getContainerContent(this.meta.containers[name]);
+                metaContainer[name] = contentJson;
+                deCompressedContainerSize = deCompressedContainerSize + JSON.stringify(contentJson).length;
+
+                this.containers[name] = Container.typeFactory(contentJson.container_type, this, name, contentJson);
             }
-            // TODO: Check Vault Version number and apply migrations if necessary
+
+            logger.debug(`saving rate: ${(1 - compressedContainerSize / deCompressedContainerSize) * 100}`);
```

Comment:
I totally agree with you on this. I kept this for the purpose of following the compression rate in the long run since as per my tests it seems to have an asymptotic shape type `squareRoot(x)` 
[https://docs.google.com/spreadsheets/d/1SxYdP_i_xBMLrOI_7uQXl_hPSBnQm-Xpi1-Mk-xH7xA/edit#gid=1129331270](url)
For which I do not know whether it might decrease for very large vault changes or not.